hw-36.ts

```diff
@@ -0,0 +1,165 @@
+        production: "Lucasfilm Ltd.",
+    },
+    {
+        id: 4,
+        title: "Harry Potter and the Half-Blood Prince",
+        year: 2009,
+        released: "15 Jul 2009",
+        runtime: "153 min",
+        genre: ["Action", "Adventure", "Family"],
+        director: "David Yates",
+        writer: "Steve Kloves, J.K. Rowling",
+        actors: ["Daniel Radcliffe", "Emma Watson", "Rupert Grint"],
+        plot: "As Harry Potter begins his sixth year at Hogwarts, he discovers an old book marked as 'the property of the Half-Blood Prince' and begins to learn more about Lord Voldemort\'s dark past.",
+        country: "United Kingdom",
+        poster: "https://m.media-amazon.com/images/M/MV5BNzU3NDg4NTAyNV5BMl5BanBnXkFtZTcwOTg2ODg1Mg@@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 492245,
+        boxOffice: "$302,305,431",
+        production: "Heyday Films, Warner Bros.",
+    },
+    {
+        id: 5,
+        title: "Harry Potter and the Sorcerer's Stone",
+        year: 2001,
+        released: "16 Nov 2001",
+        runtime: "152 min",
+        genre: ["Adventure", "Family", "Fantasy"],
+        director: "Chris Columbus",
+        writer: "J.K. Rowling, Steve Kloves",
+        actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+        plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+        country: "United Kingdom, United States",
+        poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 684604,
+        boxOffice: "$318,087,620",
+        production: "1492 Pictures, Heyday Films, Warner Brothers",
+    }
+
+]
+
+// 1. Все жанры фильмов без повторения
+const getGenres = (array: Movie[]) => {
+    const arrayOfAllGenres = array.map(movie => movie.genre)
+    const allGenres = []
+    const genres = allGenres.concat(...arrayOfAllGenres)
+    const resultArray = Array.from(new Set(genres))
+    return resultArray
+}
+console.log(getGenres(movies));
+
+// 2. Массив актёров без повторения
+const getActors = (array: Movie[]) => {
+    const arrayOfAllActors = array.map(movie => movie.actors)
+    const allActors = []
+    const actors = allActors.concat(...arrayOfAllActors)
+    const resultArray = Array.from(new Set(actors))
+    return resultArray
+}
+console.log(getActors(movies));
+
+// 3. Сортировка по рейтингу по убыванию
+const sortByImdbRating = (array: Movie[]) => {
+    const sortedArray = array.sort((a, b) => b.imdbRating - a.imdbRating)
+    return sortedArray
+}
+console.log(sortByImdbRating(movies));
+
+// 4. Новый массив, где объекты фильмов состояни из полей id, title, released, plot
+const newMoviesArray = (array: Movie[]) => array.map(movie => ({id: movie.id, title: movie.title, released: movie.released, plot: movie.plot}))
```

Comment:
super!
my favorite way to do it
`movies.map(({id, title, released, plot})=>({id, title, released, plot}))`